### PR TITLE
[front] - fix: restrict system vault access exclusively to admins

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR aims at restricting system vault views to admins only.

**References:**
- https://github.com/dust-tt/tasks/issues/1351

## Risk

Low

## Deploy Plan

Deploy `front`

### CODE REVIEW

File: front/pages/w/[wId]/vaults/[vaultId]/categories/[category]/index.tsx [hunk 1]
```diff
@@ -54,12 +55,12 @@ export const getServerSideProps = withDefaultUserAuthRequirements<
     auth,
     context.query.vaultId as string
   );
-  if (!vault || !systemVault) {
+  if (!vault || !systemVault || (vault.isSystem() && !isAdmin)) {
```
[the remark this answers is not in the document]
Comment (PopDaph) [reply front/pages/w/[wId]/vaults/[vaultId]/categories/[category]/index.tsx:58]: (Goal is to not duplicate logic that system vault is for admins)

[the remark this answers is not in the document]
Comment (JulesBelveze) [reply front/pages/w/[wId]/vaults/[vaultId]/categories/[category]/index.tsx:58]: Definitely better, forgot about it! Switching to this 🙏🏼

File: front/pages/w/[wId]/vaults/[vaultId]/categories/[category]/index.tsx [hunk 2]
```diff
@@ -54,12 +55,12 @@ export const getServerSideProps = withDefaultUserAuthRequirements<
     auth,
     context.query.vaultId as string
   );
-  if (!vault || !systemVault) {
+  if (!vault || !systemVault || (vault.canRead(auth))) {
```
Comment (PopDaph) [inline front/pages/w/[wId]/vaults/[vaultId]/categories/[category]/index.tsx:58]: Typo here, we throw 404 if can read 😅 

Why do we load both vault & systemVault? To get a link to system vault?

File: front/pages/w/[wId]/vaults/[vaultId]/index.tsx [hunk 3]
```diff
@@ -29,11 +30,12 @@ export const getServerSideProps = withDefaultUserAuthRequirements<
     auth,
     context.query.vaultId as string
   );
-  if (!vault) {
+  if (!vault || (vault.canRead(auth))) {
```
Comment (PopDaph) [inline front/pages/w/[wId]/vaults/[vaultId]/index.tsx:33]: Same here, you throw not found if can read 🙈

Comment (JulesBelveze) [reply front/pages/w/[wId]/vaults/[vaultId]/index.tsx]: 🙈

**Review by PopDaph — APPROVED** [review]
LGTM, do we need to add this also on the apps pages under front/pages/w/[wId]/vaults/[vaultId] ?

**Pull request merged**